guacamole/src/main/frontend/src/app/client/directives/guacClient.js

```diff
@@ -220,6 +221,9 @@ angular.module('client').directive('guacClient', [function guacClient() {
             display.showCursor(!localCursor);
             client.sendMouseState(event.state, true);
 
+            // Broadcast the mouse event
+            $rootScope.$broadcast('guacClientMouseEvent', event);
```

Comment:
For these events to provide context for whatever handles those events, they'll need some kind of reference to the client that would receive the event.